### DISCUSSION TITLE
Error to load generic netcdf file

### DISCUSSION THREAD

**Discussion by luizbacelar** [neuralhydrology/neuralhydrology#113] [Q&A]
Hi, 

I tried to create a netcdf file following the suggestions to add a generic dataset but I guess I am still having an issue with the format. 
When I am loading the .xml I get this error:

2023-04-30 19:01:26,073: ### Device cpu will be used for training
2023-04-30 19:01:26,076: Loading basin data into xarray data set.
  0%|                                                                                             | 0/1 [00:00<?, ?it/s]
TypeError: cannot infer freq from a non-convertible index type <class 'pandas.core.indexes.numeric.Int64Index'>

I don't know if this could be an issue with the date array format. I created this script to build the netcdf. 

import netCDF4 as nc
import numpy as np
path=''
out = '01022500.nc'
ds = nc.Dataset(path+out, 'w', format='NETCDF4')
import datetime
times = [datetime.datetime(2007, 10, 17) + datetime.timedelta(hours=hour)
         for hour in range(135194)]
date = ds.createDimension('date', 135194)
date = ds.createVariable('date', 'int64', ('date',))
calendar = 'standard'
units = 'hours since 2007-10-17 01:00'
date[:] = nc.date2num(times, units=units, calendar=calendar)
vel_ms = ds.createVariable('vel_ms', 'f4', ('date',))
wh_m = ds.createVariable('wh_m', 'f4', ('date',))
dis_m3s = ds.createVariable('dis_m3s', 'f4', ('date',))
vel_ms[:] = tab[1][1:]
wh_m[:] = tab[2][1:]
dis_m3s[:] = tab[3][1:]
ds.close()

Also, could someone maybe attach here one generic netcdf that works, so maybe I could follow the same format? I could not find it elsewhere. It could be helpful if we are not working with the CAMELS format to have one generic netcdf as an example. 
Thank you!
Luiz

**Comment by gauchm** [comment]
Hi Luiz,
from the error, it looks like your date index is not recognized properly and is loaded as int instead of datetime dtype. In general, I would highly recommend to use xarray Datasets to create and store netcdf files. I find it to be way more convenient than the netCDF4 Python API.

Attached you can find a sample file that works with the genericdataset code.

[sample.zip](https://github.com/neuralhydrology/neuralhydrology/files/11370518/sample.zip)
